modules/quantcastIdSystem.js

```diff
@@ -7,11 +7,156 @@
 
 import {submodule} from '../src/hook.js'
 import { getStorageManager } from '../src/storageManager.js';
+import { triggerPixel, logInfo } from '../src/utils.js';
+import { uspDataHandler, coppaDataHandler, gdprDataHandler } from '../src/adapterManager.js';
 
 const QUANTCAST_FPA = '__qca';
+const DEFAULT_COOKIE_EXP_TIME = 392; // (13 months - 2 days)
+const PREBID_PCODE = 'p-KceJUEvXN48CE'; // Not associated with a real account
+const DOMAIN_QSERVE = 'https://pixel.quantserve.com/pixel';
+const QUANTCAST_VENDOR_ID = '11';
+const PURPOSE_DATA_COLLECT = '1';
+const PURPOSE_PRODUCT_IMPROVEMENT = '10';
+
+var clientId;
+var cookieExpTime;
 
 export const storage = getStorageManager();
 
+export function firePixel() {
+  // check for presence of Quantcast Measure tag _qevent obj
+  if (!window._qevents) {
+    const gdprPrivacyString = gdprDataHandler.getConsentData();
+    const usPrivacyString = uspDataHandler.getConsentData();
+
+    var fpa = storage.getCookie(QUANTCAST_FPA);
+    var fpan = '0';
+    var now = new Date();
+    var domain = quantcastIdSubmodule.findRootDomain();
+    var et = now.getTime();
+    var tzo = now.getTimezoneOffset();
+    var usPrivacyParamString = '';
+    var firstPartyParamStrings;
+    var gdprParamStrings;
+
+    if (!(hasGDPRConsent(gdprPrivacyString) && hasCCPAConsent(usPrivacyString))) {
+      var expired = new Date(0).toUTCString();
+      fpan = 'u';
+      fpa = '';
+      storage.setCookie(QUANTCAST_FPA, fpa, expired, '/', domain, null);
```

Comment:
need to return here and not continue with quantserve API fire once we've expired the cookie